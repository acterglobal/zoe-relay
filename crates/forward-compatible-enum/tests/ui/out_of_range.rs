use forward_compatible_enum::ForwardCompatibleEnum;

#[derive(ForwardCompatibleEnum)]
#[forward_compatible(range = "10..20")]
pub enum OutOfRange {
    #[discriminant(5)]  // Outside range!
    Variant1,
    #[discriminant(15)]
    Variant2,
    Unknown { discriminant: u32, data: Vec<u8> },
}