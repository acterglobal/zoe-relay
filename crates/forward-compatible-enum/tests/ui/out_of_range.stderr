error: discriminant 5 is outside allowed range 10..20
 --> tests/ui/out_of_range.rs:6:5
  |
6 | /     #[discriminant(5)]  // Outside range!
7 | |     Variant1,
  | |____________^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/ui/out_of_range.rs:11:2
   |
11 | }
   |  ^ consider adding a `main` function to `$DIR/tests/ui/out_of_range.rs`
